aws-gitlab-main.yml

```diff
@@ -0,0 +1,214 @@
+Resources:
+
+  #ROLE W/ IMBEDDED POLICY
+  GitLabS3Access: 
+    Type: 'AWS::IAM::Role'
+    Properties:
+      AssumeRolePolicyDocument: 
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service:
+                - ec2.amazonaws.com
+            Action:
+              - 'sts:AssumeRole'
+      Policies:
+          - PolicyName: gl-s3-policy
+            PolicyDocument:
+              Statement:
+                - Effect: Allow
+                  Action:
+                  - 's3:AbortMultipleUpload'
+                  - 's3:CompletemultipartUpload'
+                  - 's3:ListBucket'
+                  - 's3:PutObject'
+                  - 's3:GetObject'
+                  - 's3:DeleteObject'
+                  - 's3:PutObjectAcl' 
+                  Resource:
+                    gl-s3-bucket:
+                      Type: AWS::S3::Bucket
+  GitLabS3InstanceProfile:
+    Type: 'AWS::IAM::InstanceProfile'
+    Properties:
+      Roles: 
+        - !Ref GitLabS3Access
+
+  #VPC  
+  gitlabVPC: 
+    Type: 'AWS::EC2::VPC'
+    Properties:
+      CidrBlock: 10.0.0.0/16 
+      EnableDnsHostnames: true 
+
+      #PUBLIC SUBNETS
+      gitlab-public-10.0.0.0:
+        Type: 'AWS::EC2::Subnet'
+        Properties:
+          VpcId: !Ref gitlabVPC
+          AvailabilityZone: us-west-2a 
+          CidrBlock: 10.0.0.0/24 
+          MapPublicIpOnLaunch: true 
+
+      gitlab-public-10.0.2.0:
+        Type: 'AWS::EC2::Subnet'
+        Properties:
+          VpcId: !Ref gitlabVPC
+          AvailabilityZone: us-west-2b 
+          CidrBlock: 10.0.2.0/24 
+          MapPublicIpOnLaunch: true 
+
+      #PRIVATE SUBNETS
+      gitlab-private-10.0.1.0:
+        Type: 'AWS::EC2::Subnet'
+        Properties:
+          VpcId: !Ref gitlabVPC
+          AvailabilityZone: us-west-2a 
+          CidrBlock: 10.0.1.0/24
+          MapPublicIpOnLaunch: false 
+
+      gitlab-private-10.0.3.0:
+        Type: 'AWS::EC2::Subnet'
+        Properties:
+          VpcId: !Ref gitlabVPC
+          AvailabilityZone: us-west-2b 
+          CidrBlock: 10.0.3.0/24
+          MapPublicIpOnLaunch: false 
+
+      #INTERNET GATEWAY
+      gitlab-gateway:
+        Type: 'AWS::EC2::InternetGateway'
+      
+      gitlab-GatewayAttachment:
+        Type: 'AWS::EC2::VPCGatewayAttachment'
+        Properties:
+          VpcId: !Ref gitlabVPC
+          InternetGatewayId: !Ref gitlab-gateway
+      
+      #NAT GATEWAY
+      gitlab-NAT-gateway-10.0.0.0:
+        Type: AWS::EC2::NatGateway
+        Properties: 
+          AllocationId: #not sure about this
```

Comment:
you will need to create EIP as a separate resource and reference it here using !GetAtt